corehq/apps/userreports/tests/test_app_manager_integration.py

```diff
@@ -1,17 +1,44 @@
 from __future__ import absolute_import
 from __future__ import unicode_literals
-import json
 import os
 from datetime import datetime
+
+from django.test import TestCase
 from mock import patch, Mock
-from django.test import SimpleTestCase
-from corehq.apps.app_manager.models import Application
+from corehq.apps.app_manager.models import Application, Module
+from corehq.apps.app_manager.tests.app_factory import AppFactory
 from corehq.apps.userreports.app_manager.helpers import get_case_data_sources, get_form_data_sources
 from corehq.apps.userreports.reports.builder import DEFAULT_CASE_PROPERTY_DATATYPES
 from dimagi.utils.parsing import json_format_datetime
 
 
-class AppManagerDataSourceConfigTest(SimpleTestCase):
+class AppManagerDataSourceConfigTest(TestCase):
+    domain = 'userreports_test'
+    case_type = 'app_data_case'
+    case_properties = ['first_name', 'last_name', 'children', 'dob']
+
+    @classmethod
+    def setUpClass(cls):
+        super(AppManagerDataSourceConfigTest, cls).setUpClass()
+        cls.app = Application.new_app(cls.domain, 'Application Data Source App')
+        module = cls.app.add_module(Module.new_module('Untitled Module', None))
+        module.case_type = cls.case_type
+        with open(os.path.join(os.path.dirname(__file__), 'data', 'forms', 'simple.xml')) as f:
+            form_source = f.read()
+        cls.form = cls.app.new_form(module.id, "Untitled Form", 'en', form_source)
+        AppFactory.form_requires_case(cls.form, case_type=cls.case_type, update={
+            cp: '/data/{}'.format(cp) for cp in cls.case_properties
+        })
+        cls.app.save()
```

Comment:
Does anything in these tests actually need this app to be saved now? I couldn't see any changes that would have caused DB lookups, but I might just be missing it.

Reply:
good question. it's buried deep in [export schema creation](https://github.com/dimagi/commcare-hq/blob/master/corehq/apps/export/models/new.py#L1429)